### PR TITLE
Add permission code for iOS 14

### PR DESCRIPTION
**Pull request by CaiJingLong**

### CODE REVIEW

File: lib/src/type.dart [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import 'dart:io';
```
Comment (AlexV525) [inline lib/src/type.dart:1]: ?

File: ios/Classes/PMPlugin.m [hunk 2]
```diff
@@ -84,6 +78,46 @@ - (void)onMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
...
+    BOOL auth = PHAuthorizationStatusAuthorized == status;
+    [self.manager setAuth:auth];
+    [handler reply:@(status)];
+}
+
+#if TARGET_OS_IOS
+#if __IPHONE_14_0
+
+- (void) handlePermission:(PMManager *)manager handler:(ResultHandler*) handler{
+    if (@available(iOS 14, *)) {
```
Comment (lyio) [inline ios/Classes/PMPlugin.m:91]: Is this necessary inside the `#if __IPHONE_14_0` pragma?

**Comment by yanivshaked** [comment]
Hi @CaiJingLong,

I think you should also add support for calling `PHPhotoLibrary.shared().presentLimitedLibraryPicker`, this will invoke a native popup asking the user to add more images to his selection (useful to implement the WhatsApp asset picker pattern).

See [here](https://developer.apple.com/documentation/photokit/phphotolibrary/3616113-presentlimitedlibrarypicker)

Thanks!

**Comment by CaiJingLong** [comment]
> Hi @CaiJingLong,
> 
> I think you should also add support for calling `PHPhotoLibrary.shared().presentLimitedLibraryPicker`, this will invoke a native popup asking the user to add more images to his selection (useful to implement the WhatsApp asset picker pattern).
> 
> See [here](https://developer.apple.com/documentation/photokit/phphotolibrary/3616113-presentlimitedlibrarypicker)
> 
> Thanks!

I don't know much about this mode. Is there any corresponding article / blog that I can refer to?

File: android/src/main/kotlin/top/kikt/imagescanner/core/entity/PermissionResult.kt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package top.kikt.imagescanner.core.entity
+
+enum class PermissionResult(val value: Int) {
```
Comment (AlexV525) [inline android/src/main/kotlin/top/kikt/imagescanner/core/entity/PermissionResult.kt:3]: So we don't have `Limited` in this PR?

File: lib/src/manager.dart [hunk 4]
```diff
@@ -4,13 +4,30 @@ Plugin _plugin = Plugin();
...
+    return (await requestPermissionExtend()).isAuth;
+  }
+
+  /// Android: WRITE_EXTERNAL_STORAGE, READ_EXTERNAL_STORAGE, MEDIA_LOCATION
+  ///
+  /// iOS: NSPhotoLibraryUsageDescription of info.plist
+  ///
+  /// macOS of Release.entitlements:
+  ///   - com.apple.security.assets.movies.read-write
+  ///   - com.apple.security.assets.music.read-write
```
Comment (AlexV525) [inline lib/src/manager.dart:25]:
```suggestion
  ///  - com.apple.security.assets.movies.read-write
  ///  - com.apple.security.assets.music.read-write
```

File: ios/Classes/PMPlugin.m [hunk 5]
```diff
@@ -44,15 +44,9 @@ - (void)onMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
   PMManager *manager = self.manager;
```
Comment (AlexV525) [inline ios/Classes/PMPlugin.m:44]: Can we implement permissions according to the `permission_handler` plugin? See https://github.com/Baseflow/flutter-permission-handler/blob/master/permission_handler/ios/Classes/strategies/PhotoPermissionStrategy.m

File: android/src/main/kotlin/top/kikt/imagescanner/core/entity/PermissionResult.kt (hunk 3 above)
Comment (lyio) [reply android/src/main/kotlin/top/kikt/imagescanner/core/entity/PermissionResult.kt:3]: @AlexV525 isn't `limited` an iOS only feature? This is in the Android plugin code, isn't it?

Comment (AlexV525) [reply android/src/main/kotlin/top/kikt/imagescanner/core/entity/PermissionResult.kt:3]: > This is in the Android plugin code, isn't it?

Yes you're right. 👍

**Comment by CaiJingLong** [comment]
@AlexV525 
Do you think the reading and writing of IOS should be separated?

**Comment by AlexV525** [comment]
> Do you think the reading and writing of IOS should be separated?

A separate PR to improve permission handlers might be more appropriate. But regarding the read/write, yes and not only on iOS, the Android's read/write might need to separate too. Let's discuss this later.

**Comment by yanivshaked** [comment]
> I don't know much about this mode. Is there any corresponding article / blog that I can refer to?

https://www.wwdcnotes.com/notes/wwdc20/10641/

https://www.hackingwithswift.com/forums/swiftui/how-to-call-phphotolibrary-s-presentlimitedlibrarypicker-from-swiftui/3747

**Comment by CaiJingLong** [comment]
> > Do you think the reading and writing of IOS should be separated?
> 
> A separate PR to improve permission handlers might be more appropriate. But regarding the read/write, yes and not only on iOS, the Android's read/write might need to separate too. Let's discuss this later.

Read and write permissions are the same permission group in Android.

**Comment by CaiJingLong** [comment]
> > I don't know much about this mode. Is there any corresponding article / blog that I can refer to?
> 
> https://www.wwdcnotes.com/notes/wwdc20/10641/
> 
> https://www.hackingwithswift.com/forums/swiftui/how-to-call-phphotolibrary-s-presentlimitedlibrarypicker-from-swiftui/3747

OK, get it. I'll open new branch to do it.

**Comment by CaiJingLong** [comment]
> > I don't know much about this mode. Is there any corresponding article / blog that I can refer to?
> 
> https://www.wwdcnotes.com/notes/wwdc20/10641/
> 
> https://www.hackingwithswift.com/forums/swiftui/how-to-call-phphotolibrary-s-presentlimitedlibrarypicker-from-swiftui/3747

OK, get it. I'll open new branch to do it.

**Pull request merged**